src/js/tabs/auxiliary.js

```diff
@@ -13,6 +13,156 @@ import inflection from "inflection";
+    { buildKey: 'servos',    buildOption: ['USE_SERVOS']},
+    { buildKey: 'telemetry', buildOption: ['USE_TELEMETRY', 'USE_TELEMETRY_SMARTPORT']},
+    { buildKey: 'vtx',       buildOption: ['USE_VTX']},
+];
+
+const flightModes = ["ARM","ANGLE","HORIZON","ANTI GRAVITY","MAG","HEADFREE","HEADADJ","SERVO1","SERVO2","SERVO3",
+                     "FAILSAFE","AIR MODE","FPV ANGLE MIX","FLIP OVER AFTER CRASH","USER1","USER2","USER3","USER4","ACRO TRAINER","LAUNCH CONTROL"];
+
+// Categories to be mapped with buildMap. Category 'all' are virtuel and always included
+const categoryTable = [
+    { name: '3D',         buildKey: ['dshot'],     modes: ['3D', '3D DISABLE / SWITCH']},
+    { name: 'BEEP',       buildKey: ['all'],       modes: ['BEEPERON', 'BEEPER', 'BEEPER MUTE', 'GPS BEEP SATELLITE COUNT']},
+    { name: 'BLACKBOX',   buildKey: ['all'],       modes: ['BLACKBOX', 'BLACKBOX ERASE']},
+    { name: 'CAM',        buildKey: ['cam'],       modes: ['CAMERA CONTROL 1', 'CAMERA CONTROL 2', 'CAMERA CONTROL 3']},
+    { name: 'FLIGHTMODE', buildKey: ['all'],       modes: flightModes},
+    { name: 'GPS',        buildKey: ['gps'],       modes: ['GPS BEEP SATELLITE COUNT', 'GPS RESCUE']},
+    { name: 'LED',        buildKey: ['led_strip'], modes: ['LEDLOW']},
+    { name: 'OSD',        buildKey: ['osd'],       modes: ['OSD DISABLE']},
+    { name: 'OTHER',      buildKey: ['all'],       modes: ['CALIB', 'MSP OVERRIDE', 'LAP TIMER RESET', 'PASSTHRU', 'PARALYZE', 'PID AUDIO', 'PREARM']},
+    { name: 'SERVO',      buildKey: ['servos'],    modes: ['SERVO1', 'SERVO2', 'SERVO3']},
+    { name: 'TELEMETRI',  buildKey: ['telemetry'], modes: ['TELEMETRY']},
+    { name: 'USER',       buildKey: ['all'],       modes: ['USER1', 'USER2', 'USER3', 'USER4']},
+    { name: 'VTX',        buildKey: ['vtx'],       modes: ['STICK COMMANDS DISABLE', 'VTX CONTROL DISABLE', 'VTX PIT MODE']},
+];
+
+let modeList = [];
+
+function inBuildMap(map, name) {
+    if (name == 'all') {
+        return true;
+    }
+    for (let value of map) {
+        if (name == value.buildKey) {
+            for (let option of value.buildOption) {
+                if (FC.CONFIG.buildOptions.includes(option)) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+function isSelectedMode(mList, modeName) {
+    for (let value of mList) {
+        if (value.includes(modeName)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function resolveCategoryName(category, choise) {
+    let mList = [];
+    for (let value of choise) {
+        for (let elm of category) {
+            if (value == elm.name) {
+                mList.push(elm.modes);
+            }
+        }
+    }
+    return mList;
+}
+
+function isPreSelectedCategory(categoryList, categoryName) {
+    for (let value of categoryList) {
+        if (value == categoryName) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
Suggested below, you don't really need this, can use [.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

```suggestion
```